ehr/src/org/labkey/ehr/EHRController.java

```diff
@@ -1582,13 +1591,18 @@ public Object execute(PopulateLookupsForm form, BindException errors) throws Exc
 
                     if (null != _additionalReportsResource)
                     {
+                        // Merge in additional reports (based on report name) with replace option
                         responseText.append("Populating additional reports from " + _additionalReportsModule.getName() + " module\n");
                         if (behaviorType != null && behaviorType != AuditBehaviorType.NONE)
                             auditEvent = createTransactionAuditEvent(getContainer(), QueryService.AuditAction.INSERT);
 
+                        context = new DataIteratorContext(batchErrors);
+                        context.setConfigParameters(configParameters);
+                        context.setInsertOption(QueryUpdateService.InsertOption.REPLACE);
+                        context.setAlternateKeys(Set.of("reportname"));
```

Comment:
Does this work as desired in dev machines that may have multiple EHR containers with reports loaded? There may be multiple existing rows with the same `reportname` - does this operate on just the ones in the current container? And should we have a constraint on `container, reportname`?

Reply:
Yes this respects the container, so it won't replace a report in another EHR container. 

I would like to add that constraint but it looks like at least ONPRC has multiple reports with the same name. They are not using this action to populate reports so it won't be an issue for them, but they are using the ehr.reports table so the constraint would have a conflict if we tried to add it to the table. I can work with them to rename those reports to unique names but that may take a while to get done.